File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Embeddings/EmbeddingsResponse.cs
namespace Microsoft.Teams.AI.AI.Embeddings
{
    /// <summary>
    /// Response returned by a `EmbeddingsClient`.
    /// </summary>
    public class EmbeddingsResponse
    {
        /// <summary>
        /// Status of the embeddings response.
        /// </summary>
        public EmbeddingsResponseStatus Status { get; set; }

        /// <summary>
        /// Optional. Embeddings for the given inputs.
        /// </summary>
        public IList<ReadOnlyMemory<float>>? Output { get; set; }

        /// <summary>
        /// Optional. Message when status is not equal to `success`.
        /// </summary>
        public string? Message { get; set; }
    }

    /// <summary>
    /// Status of the embeddings response.
    /// </summary>
    public enum EmbeddingsResponseStatus
    {
        /// <summary>
        /// The embeddings were successfully created.
        /// </summary>
        Success,

        /// <summary>
        /// An error occurred while creating the embeddings.
        /// </summary>
        Failure,

        /// <summary>
        /// The request was rate limited.
        /// </summary>
        RateLimited,
    }
}

File: python/packages/ai/teams/ai/embeddings/embeddings_response.py
"""
Copyright (c) Microsoft Corporation. All rights reserved.
Licensed under the MIT License.
"""

from __future__ import annotations

from dataclasses import dataclass
from typing import List, Literal, Optional, Union

EmbeddingsResponseStatus = Literal[
    "success",  # The embeddings were successfully created.
    "error",  # An error occurred while creating the embeddings.
    "rate_limited",  # The request was rate limited.
]


@dataclass
class EmbeddingsResponse:
    """
    Response returned for embeddings.
    """

    status: EmbeddingsResponseStatus
    "Status of the embeddings response."

    output: Optional[Union[List[List[float]], str]] = None
    "Optional. Embeddings for the given inputs or the error string."

    message: Optional[str] = None
    "Optional. Message status."
